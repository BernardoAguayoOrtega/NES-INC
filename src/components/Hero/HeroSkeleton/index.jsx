// import react
import React from 'react';
//import react-loading-skeleton
import Skeleton, { SkeletonTheme } from 'react-loading-skeleton';

const HeroSkeleton = () => {
  return (
    <SkeletonTheme color="#36404e" highlightColor="#2E2E32">
      <div
        aria-label="hero content"
        className={'home__hero-container'}
      >
        <Skeleton width={1200} height={320}/>
      </div>
    </SkeletonTheme> 
  )
};

export default HeroSkeleton;
