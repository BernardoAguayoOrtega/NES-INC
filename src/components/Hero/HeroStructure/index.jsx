// import react
import React from 'react';

const HeroStructure = ({ background, caption, type = 'large' }) => {
  return (
    <div
      aria-label="hero content"
      className={`home__hero-container ${type === 'small' && 'small'}`}
    >
      <h1 aria-label="hero title" className="home__hero-caption">
        {caption}
      </h1>
      <figure aria-label="hero figure" className="home__hero-figure">
        <img
          aria-label="hero image"
          className="home__hero-image"
          src={background}
          alt={`${caption} image`}
        />
      </figure>
    </div>
  )
}

export default HeroStructure;
