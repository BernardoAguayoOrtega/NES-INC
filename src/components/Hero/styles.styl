.home__hero-container
 background: red;