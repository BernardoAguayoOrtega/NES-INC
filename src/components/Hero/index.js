// import react
import React from 'react';
// use styles
import './styles.styl';

// create and export Hero component
const Hero = ({ background, caption }) => (
  <div className="home__hero-container">
    <figure className="home__hero-figure">
      <img className="home__hero-image" src={background} />
    </figure>
    <h1 className="home__hero-caption">{caption}</h1>
  </div>
);

export default Hero;
