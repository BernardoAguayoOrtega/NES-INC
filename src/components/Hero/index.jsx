// import react
import React, { useState, useEffect } from 'react';
// import Hero Structure
import HeroStructure from './HeroStructure/index.jsx'
// import HeroSkeleton
import HeroSkeleton from './HeroSkeleton/index.jsx'
// use styles
import './styles.styl';

// create and export Hero component
const Hero = ({ background, caption }) => {
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    setTimeout(() => {
      setLoading(false);
    }, 2000);
  }, [])

  return loading ? <HeroSkeleton /> : <HeroStructure background={background} caption={caption} />;
}

export default Hero;
