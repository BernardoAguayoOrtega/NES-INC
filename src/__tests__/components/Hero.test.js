// import react
import React from 'react';
// import shallow
import { shallow } from 'enzyme';
// import toJson
import toJson from 'enzyme-to-json';
// import app component
import Hero from '../../components/Hero/index.jsx';

describe('<Hero type=small/>', () => {
  const wrapper = shallow(<Hero type="small" />);

  it('is render the Hero?', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('are there a container?', () => {
    expect(wrapper.find('.home__hero-container').exists()).toBe(true);
  });

  it('are container have two children?', () => {
    expect(wrapper.find('.home__hero-container').children().length).toBe(2);
  });

  it('should have home hero class container?', () => {
    expect(wrapper.hasClass('home__hero-container')).toBe(true);
  });

  it('matches the snap shot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('<Hero />', () => {
  const wrapper = shallow(<Hero />);

  it('are default type works?', () => {
    expect(wrapper.exists()).toBeTruthy()
  })
})
