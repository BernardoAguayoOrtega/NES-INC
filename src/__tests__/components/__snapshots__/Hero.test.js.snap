// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero type=small/> matches the snap shot 1`] = `
<div
  aria-label={true}
  className="home__hero-container small"
>
  <h1
    className="home__hero-caption"
  />
  <figure
    className="home__hero-figure"
  >
    <img
      className="home__hero-image"
    />
  </figure>
</div>
`;
